pymedphys/_dicom/delivery/core.py

```diff
@@ -37,15 +45,72 @@ def _pretty_print(string):
     return "\n\n".join(wrapped)
 
 
+def check_for_supported_collimation_device(beam_limiting_device_sequence):
+    rt_beam_limiting_device_types = {
+        item.RTBeamLimitingDeviceType for item in beam_limiting_device_sequence
+    }
+
+    supported_configurations = [{"MLCX", "ASYMY"}]
+
+    if not rt_beam_limiting_device_types in supported_configurations:
+        raise NotSupportedError(
+            _pretty_print(
+                """\
+                        Currently only DICOM files where the beam
+                        limiting devices consist of one of the following
+                        combinations are supported:
+
+                        * {supported_configurations}
+
+                        The provided RT Plan DICOM file has the
+                        following:
+
+                            {rt_beam_limiting_device_types}
+
+                        This is not yet supported.
+                        This is due to a range of assumptions being made
+                        internally that assume a single jaw system.
+                        There are some cases where this restriction is
+                        too tight. Currently however there is not enough
+                        testing data to appropriately implement these
+                        cases.
+                        If you would like to have your device supported
+                        please consider uploading anonymised DICOM files
+                        and their TRF counterparts to the following
+                        issue
+                        <https://github.com/pymedphys/pymedphys/issues/1142>.
+
+                        If you know what you're doing, and are willing
+                        to accept potentially ignoring a collimation
+                        device you may pass `device_strict=False` to
+                        `pymedphys.Delivery.from_dicom`.
+
+                    """
+            ).format(
+                supported_configurations="\n* ".join(
+                    [str(item) for item in supported_configurations]
+                ),
+                rt_beam_limiting_device_types=rt_beam_limiting_device_types,
+            )
+        )
+
+
 class DeliveryDicom(DeliveryBase):
     @classmethod
-    def from_dicom(cls, dicom_dataset, fraction_number=None):
+    def from_dicom(
+        cls, rtplan: dicom_path_or_dataset, fraction_number=None, device_strict=True
+    ):
+        if isinstance(rtplan, pydicom.Dataset):
```

Comment:
this seems like an incredibly useful few lines of code (pathlike | object one would deserialise from a file having said path).  While I don't think it's appropriate at the moment to extract it, I believe we should keep our eyes open for other places where it would be very useful to see just what level of generalization of the concept is appropriate.

Reply:
Yup I agree.